public/components/trace_analytics/components/traces/traces_content.tsx

```diff
@@ -78,12 +83,31 @@ export function TracesContent(props: TracesProps) {
       processTimeStamp(endTime, mode),
       page
     );
-    await handleTracesRequest(http, DSL, timeFilterDSL, tableItems, setTableItems, mode, sort);
+    await handleTracesRequest(
+      http,
+      DSL,
+      timeFilterDSL,
+      tableItems,
+      setTableItems,
+      mode,
+      props.dataSourceMDSId[0].id,
+      sort
+    );
     setLoading(false);
   };
 
   return (
     <>
+      {props.dataSourceEnabled && (
+        <DataSourceMenu
+          setMenuMountPoint={props.setActionMenu}
+          componentType={'DataSourceView'}
+          componentConfig={{
+            activeOption: dataSourceMDSId,
+            fullWidth: true,
+          }}
+        />
+      )}
```

Comment:
there was a DataSourceMenu in home.tsx, why is it needed again here?

Reply:
there are two menus one is readonly and one is a selector